stdlib/public/core/UnavailableStringAPIs.swift.gyb

```diff
@@ -109,3 +109,41 @@ ${stringSubscriptComment}
     Builtin.unreachable()
   }
 }
+
+% for View in ['Character', 'UnicodeScalar', 'UTF16', 'UTF8']:
+%   Index = 'String.%sView.Index' % View
+%   Distance = 'String.%sView.Index' % View
+extension ${Index} {
+  @available(
+    *, unavailable,
+    message: "To get the next index call 'index(after:)' on the instance of the ${View} string view that produced the index")
+  public func successor() -> ${Index} {
+    Builtin.unreachable()
+  }
+  @available(
+    *, unavailable,
+    message: "To get the previous index call 'index(after:)' on the instance of the ${View} string view that produced the index")
```

Comment:
Should be `index(before:)`—also, skip this for `UTF8View.Index` or have the message say that it isn't bidirectional?